exporter/alibabacloudlogserviceexporter/tracedata_to_logservice.go

```diff
@@ -15,393 +15,225 @@
 package alibabacloudlogserviceexporter
 
 import (
-	"encoding/hex"
 	"encoding/json"
 	"strconv"
 	"time"
 
 	sls "github.com/aliyun/aliyun-log-go-sdk"
-	commonpb "github.com/census-instrumentation/opencensus-proto/gen-go/agent/common/v1"
-	resourcepb "github.com/census-instrumentation/opencensus-proto/gen-go/resource/v1"
-	tracepb "github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1"
 	"github.com/gogo/protobuf/proto"
-	"go.opentelemetry.io/collector/consumer/consumerdata"
+	"go.opentelemetry.io/collector/consumer/pdata"
 	tracetranslator "go.opentelemetry.io/collector/translator/trace"
-	"google.golang.org/protobuf/types/known/timestamppb"
 )
 
 const (
 	traceIDField       = "traceID"
 	spanIDField        = "spanID"
 	parentSpanIDField  = "parentSpanID"
-	operationNameField = "operationName"
-	referenceField     = "reference"
-	startTimeField     = "startTime"
+	nameField          = "name"
+	kindField          = "kind"
+	linksField         = "links"
+	timeField          = "time"
+	startTimeField     = "start"
+	endTimeField       = "end"
+	traceStateField    = "traceState"
 	durationField      = "duration"
-	tagsPrefix         = "tags."
+	attributeField     = "attribute"
+	statusCodeField    = "statusCode"
+	statusMessageField = "statusMessage"
 	logsField          = "logs"
-	serviceNameField   = "process.serviceName"
-	processTagsPrefix  = "process.tags."
-)
-
-const (
-	// Tags
-	opencensusLanguage        = "language"
-	opencensusExporterVersion = "exporter_version"
-	opencensusCoreLibVersion  = "core_lib_version"
-	opencensusResourceType    = "resource_type"
 )
 
 // traceDataToLogService translates trace data into the LogService format.
-func traceDataToLogServiceData(td consumerdata.TraceData) []*sls.Log {
-	logs := spansToLogServiceData(td.Spans)
-	tagContents := nodeAndResourceToLogContent(td.Node, td.Resource)
-
-	for _, log := range logs {
-		log.Contents = append(log.Contents, tagContents...)
-	}
-	return logs
+func traceDataToLogServiceData(td pdata.Traces) ([]*sls.Log, int) {
+	var slsLogs []*sls.Log
+	resourceSpansSlice := td.ResourceSpans()
+	for i := 0; i < resourceSpansSlice.Len(); i++ {
+		resourceSpans := resourceSpansSlice.At(i)
+		if resourceSpans.IsNil() {
+			continue
```

Comment:
Should add a counter for dropped spans

Reply:
resource span is never nil when used in production, so the nil should not be counted